sklearnex/linear_model/linear.py

```diff
@@ -156,8 +186,14 @@ def _onedal_fit_supported(self, method_name, *data):
         n_samples = _num_samples(X)
         n_features = _num_features(X, fallback_1d=True)
 
-        # Check if equations are well defined
+        # Note: support for some variants was either introduced in oneDAL 2025.1,
+        # or had bugs in some uncommon cases in older versions.
         is_underdetermined = n_samples < (n_features + int(self.fit_intercept))
+        supports_all_variants = daal_check_version((2025, "P", 1))
+        if hasattr(y, "shape"):

```

Comment:
Why is this here?

Reply:
Different behavior or `_num_feature` with lists and arrays:
https://github.com/intel/scikit-learn-intelex/pull/2083#discussion_r1810369965

Reply:
https://github.com/intel/scikit-learn-intelex/blob/f350c0d183f3a2503ad9a09dd36180d68cef713b/onedal/utils/validation.py#L379 is definitely the cause.  We can change it in this PR or we can make a small bugfix and run it against CI. Your choice!

Reply:
Looks like changing `_num_features` to have the desierd behavior didn't break any other tests, so I'm modifying it here.